site/content/docs/user-guide/setup/install.md

```diff
@@ -24,10 +24,10 @@ Krew self-hosts).
       set -x; cd "$(mktemp -d)" &&
       OS="$(uname | tr '[:upper:]' '[:lower:]')" &&
       ARCH="$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/')" &&
-      curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.tar.gz" &&
-      tar zxvf krew.tar.gz &&
-      KREW=./krew-"${OS}_${ARCH}" &&
-      "$KREW" install krew
+      KREWNAME="krew-${OS}_${ARCH}" &&
+      curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/latest/download/${KREWNAME}.tar.gz" &&
+      tar zxvf "${KREWNAME}.tar.gz" &&
+      ./"${KREWNAME}" install --manifest-url https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.yaml --archive ./${KREWNAME}.tar.gz
```

Comment:
```suggestion
      ./"${KREWNAME}" install krew
```

The `--manifest` and `--archive` options are only for testing. When a plugin is installed like this, it will be in a detached state where it is not updated when new versions are published on `krew-index`. So we _do_ need to download twice here.

Reply:
Arf, too bad.
Maybe would be possible to allow `--archive` without `--manifest[-url]` to use local archive?

Reply:
That does not work either, I'm afraid. It should not be a big deal though. Installation is only done once, and you are already cutting down the download size.

Reply:
Looks like it works, by removing the lines doing the check in cmd/krew/cmd/install.go and compiling.

```diff
diff --git i/cmd/krew/cmd/install.go w/cmd/krew/cmd/install.go
index a574346..ac4b265 100644
--- i/cmd/krew/cmd/install.go
+++ w/cmd/krew/cmd/install.go
@@ -96,10 +96,6 @@ Remarks:
                                return errors.New("must specify either specify either plugin names (via positional arguments or STDIN), or --manifest/--manifest-url; not both")
                        }
 
-                       if *archiveFileOverride != "" && *manifest == "" && *manifestURL == "" {
-                               return errors.New("--archive can be specified only with --manifest or --manifest-url")
-                       }
-
                        var install []pluginEntry
                        for _, name := range pluginNames {
                                indexName, pluginName := pathutil.CanonicalPluginName(name)
```

```console
$ curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.tar.gz"
$ ./out/bin/krew-linux_amd64 install krew --archive ./krew.tar.gz
Adding "default" plugin index from https://github.com/kubernetes-sigs/krew-index.git.
Updated the local copy of plugin index.
Installing plugin: krew
Installed plugin: krew
\
 | Use this plugin:
 | 	kubectl krew
 | Documentation:
 | 	https://krew.sigs.k8s.io/
 | Caveats:
 | \
 |  | krew is now installed! To start using kubectl plugins, you need to add
 |  | krew's installation directory to your PATH:
 |  | 
 |  |   * macOS/Linux:
 |  |     - Add the following to your ~/.bashrc or ~/.zshrc:
 |  |         export PATH="${KREW_ROOT:-$HOME/.krew}/bin:$PATH"
 |  |     - Restart your shell.
 |  | 
 |  |   * Windows: Add %USERPROFILE%\.krew\bin to your PATH environment variable
 |  | 
 |  | To list krew commands and to get help, run:
 |  |   $ kubectl krew
 |  | For a full list of available plugins, run:
 |  |   $ kubectl krew search
 |  | 
 |  | You can find documentation at
 |  |   https://krew.sigs.k8s.io/docs/user-guide/quickstart/.
 | /
/

$ tail ~/.krew/receipts/krew.yaml
      matchLabels:
        arch: amd64
        os: windows
    sha256: a26deea175f70264260d59a4e061778a892f8a8e301ac261660dd7d24c551c99
    uri: https://github.com/kubernetes-sigs/krew/releases/download/v0.4.1/krew.tar.gz
  shortDescription: Package manager for kubectl plugins.
  version: v0.4.1
status:
  source:
    name: default
```

Source is not 'detached'.
And I can confirm tat there is no download of archive through the install.

Reply:
I think we reserved those arguments specifically for local testing so they are used for installing "detached" plugins. I’m not sure if it's worth changing that just to prevent downloading a small tarball once more.